src/dpp/channel.cpp

```diff
@@ -78,6 +78,61 @@ std::string channel::get_mention() const {
 	return "<#" + std::to_string(id) + ">";
 }
 
+channel& channel::set_name(const std::string& name) {
+	this->name = utility::validate(name, 1, 100, "name must be at least 1 character");
+	return *this;
+}
+
+channel& channel::set_topic(const std::string& topic) {
+	this->topic = utility::utf8substr(topic, 0, 1024);
+	return *this;
+}
+
+channel& channel::set_parent_id(const snowflake parent_id) {
```

Comment:
do we need methods for these,.when we can do c->parent_id = x ?

Reply:
I mean that logic can be applied to every other class but we still have setters, don't we?